File: bin/validate.py
#!/usr/bin/env python3

import os
import sys
import json
import argparse
import jsonschema

def validate(datafile, schemafile):
    data = json.load(open(datafile))
    schema = json.load(open(schemafile))

    jsonschema.validate(data, schema)

parser = argparse.ArgumentParser(description='Validate the JSON for a form.')
parser.add_argument('datafile', metavar='F', help='JSON file to validate')
parser.add_argument('schemafile', metavar='S', help='Schema file to validate against')

args = parser.parse_args()
datafile = args.datafile
schemafile = args.schemafile

if (not os.path.isfile(datafile)):
    print("{} does not exist".format(datafile), file=sys.stderr)
    exit(1)

if (not os.path.isfile(schemafile)):
    print("{} does not exist".format(schemafile), file=sys.stderr)
    exit(1)

validate(datafile, schemafile)

File: bin/validate.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const Ajv = require('ajv');
const ajv = new Ajv();

function validate(datafile, schemafile) {
  const opts = { 'encoding': 'utf8' };
  const datastr = fs.readFileSync(datafile, opts); 
  const schemastr = fs.readFileSync(schemafile, opts);

  let valid = ajv.validate(JSON.parse(schemastr), JSON.parse(datastr));
  if (!valid) {
    process.stderr.write(`${ajv.errorsText()}\n`);
    process.exit(1);
  }
  else {
    process.exit();  
  }
};

const argv = require('yargs')
            .usage('Usage: validate.py [-h] [-f] <file> [-s] <schema>')
            .example('$0 -f examples/apply-for-a-medal.json -s schemas/form.json', 'Validate the given JSON file against the given schema.')
            .option('file', {
              'alias': 'f',
              'string': true,
              'requiresArg': true,
              'nargs': 1,
              'describe': 'JSON file to validate.'
            })
            .option('schema', {
              'alias': 's',
              'string': true,
              'requiresArg': true,
              'nargs': 1,
              'describe': 'JSON Schema file to validate against'
            })
            .demandOption(['file', 'schema'], 'Please provide a JSON file to validated and a JSON Schema file to validate against.')
            .help()
            .alias('help', 'h')
            .argv

datafile = argv.file;
schemafile = argv.schema;

validate(datafile, schemafile);
